iothub/service/src/Jobs/ScheduledJobsClient.cs

```diff
@@ -0,0 +1,364 @@
+
+namespace Microsoft.Azure.Devices
+{
+    /// <summary>
+    /// Scheduled jobs management.
+    /// </summary>
+    /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-jobs"/>.
+    public class ScheduledJobsClient
+    {
+        private readonly string _hostName;
+        private readonly IotHubConnectionProperties _credentialProvider;
+        private readonly HttpClient _httpClient;
+        private readonly HttpRequestMessageFactory _httpRequestMessageFactory;
+
+        private const string JobsUriFormat = "/jobs/v2/{0}";
+        private const string JobsQueryFormat = "/jobs/v2/query";
+        private const string CancelJobUriFormat = "/jobs/v2/{0}/cancel";
+
+        private const string ContinuationTokenHeader = "x-ms-continuation";
+        private const string PageSizeHeader = "x-ms-max-item-count";
+
+        private static readonly TimeSpan s_defaultOperationTimeout = TimeSpan.FromSeconds(100);
+
+        /// <summary>
+        /// Creates client, provided for unit testing purposes only.
+        /// </summary>
+        protected ScheduledJobsClient()
+        {
+        }
+
+        internal ScheduledJobsClient(string hostName, IotHubConnectionProperties credentialProvider, HttpClient httpClient, HttpRequestMessageFactory httpRequestMessageFactory)
+        {
+            _hostName = hostName;
+            _credentialProvider = credentialProvider;
+            _httpClient = httpClient;
+            _httpRequestMessageFactory = httpRequestMessageFactory;
+        }
+
+        /// <summary>
+        /// Gets the job with the specified Id.
+        /// </summary>
+        /// <param name="jobId">Id of the job to get.</param>
+        /// <param name="cancellationToken">Task cancellation token.</param>
+        /// <returns>The matching JobResponse object.</returns>
+        /// <exception cref="ArgumentNullException">Thrown when the provided <paramref name="jobId"/> is null.</exception>
+        /// <exception cref="ArgumentException">Thrown if the <paramref name="jobId"/> is empty or white space.</exception>
+        /// <exception cref="IotHubException">
+        /// Thrown if IoT hub responded to the request with a non-successful status code. For example, if the provided
+        /// request was throttled, <see cref="IotHubThrottledException"/> is thrown. For a complete list of possible
+        /// error cases, see <see cref="Common.Exceptions"/>.
+        /// </exception>
+        /// <exception cref="HttpRequestException">
+        /// If the HTTP request fails due to an underlying issue such as network connectivity, DNS failure, or server
+        /// certificate validation.
+        /// </exception>
+        /// <exception cref="OperationCanceledException">If the provided <paramref name="cancellationToken"/> has requested cancellation.</exception>
+        public virtual async Task<JobResponse> GetAsync(string jobId, CancellationToken cancellationToken = default)
```

Comment:
I don't love the name "JobResponse" in this context. Maybe "ScheduledJob" so that the API reads like:

```csharp
ScheduledJob scheduledJob = await serviceClient.ScheduledJobs.GetAsyc(...);
```